// base/lib/gulp/config.js
/**
 * config.js
 * Created by raul on 11/20/14.
 */

var dest = "public/";
var src = 'app/';

module.exports = {
	dest:dest,
	assets: {
		sourceRoot: src + 'assets/',
		src: src + 'assets/**/*',
		dest: dest
	},
	jade: {
		sourceRoot : src + "views/",
		src: src + "views/*.jade",
		dest: dest,
		config: src + 'views/config.json'
	},
	templates: {
		sourceRoot : src + "templates/",
		src: src + 'templates/**/*.hbs',
		dest: dest + 'js/',
		outputName: 'templates.js',
	},
	typescript: {
		sourceRoot : src + "scripts/",
		src: src + 'scripts/**/*.ts',
		dest: dest + 'js/',
		outputName: 'main.js',
		tsProject: {
			settings: {
				target:'ES5',
				declarationFiles: false,
				noExternalResolve: false,
				sortOutput: true
			}
		}
	},
	styles: {
		sourceRoot : src + "styles/",
		src: src + 'styles/*.{sass,scss}',
		dest: dest + 'css/',
		outputName: 'main.css'
	},
	vendor: {
		sourceRoot : "vendor/",
		js: {
			sourceRoot : "vendor/javascripts/",
			src: 'vendor/javascripts/vendor.js',
			dest: dest + 'js/',
			outputName: 'vendor.js'
		},
		styles: {
			sourceRoot : "vendor/stylesheets/",
			src: 'vendor/stylesheets/vendor.css',
			dest: dest + 'css/'
		}
	},
	browserSync : {
		//proxy: "192.168.33.174",
		server: {
            baseDir: "public",
            index: "index.html"
        }
	}
};
